winter/query_parameter.py

```diff
@@ -34,29 +36,52 @@ def get_query_param_mapping(method: ComponentMethod, map_to: str) -> Optional[st
 
 class QueryParameterResolver(ArgumentResolver):
 
+    def __init__(self):
+        super().__init__()
+        self._cache = {}
+
     def is_supported(self, argument: ComponentMethodArgument) -> bool:
-        return get_query_param_mapping(argument.method, argument.name) is not None
+        if argument not in self._cache:
+            self._set_cache(argument)
+        return self._cache[argument] is not None
 
     def resolve_argument(self, argument: ComponentMethodArgument, http_request: rest_framework.request.Request):
         query_parameters = http_request.query_params
-        query_parameter_name = get_query_param_mapping(argument.method, argument.name)
-        if query_parameter_name not in query_parameters:
-            if argument.has_default():
-                return argument.default
-            elif not argument.required:
-                return None
-            else:
-                raise ParseError(f'Missing required query parameter "{query_parameter_name}"')
-        if type_utils.is_iterable(argument.type_):
-            query_parameter_value = query_parameters.getlist(query_parameter_name)
-        else:
-            query_parameter_value = query_parameters[query_parameter_name]
 
-        @pydantic.dataclasses.dataclass
-        class FieldData:
-            value: argument.type_
+        parameter_name, pydantic_data_class, is_iterable = self._cache[argument]
```

Comment:
It shouldn't be necessary to call `is_supported` before calling `resolve_argument`. I think it's a good reason to split `is_supported` and `resolve_argument` methods tests.